app/src/main/java/in/avimarine/boatangels/activities/MainActivity.java

```diff
@@ -48,6 +49,7 @@ protected void onCreate(Bundle savedInstanceState) {
     setContentView(R.layout.activity_main);
     ButterKnife.bind(this);
     FirebaseAuth auth = FirebaseAuth.getInstance();
+    FirebaseUser CurrentUser = auth.getCurrentUser();
```

Comment:
This variable is not used.
The next if should have reference it and not call auth.getCurrentUser again of you could have not used this var at all.